source/extensions/transport_sockets/proxy_protocol/proxy_protocol.h

```diff
@@ -52,6 +53,9 @@ class UpstreamProxyProtocolSocket : public TransportSockets::PassthroughSocket,
   const UpstreamProxyProtocolStats& stats_;
   const bool pass_all_tlvs_;
   absl::flat_hash_set<uint8_t> pass_through_tlvs_{};
```

Comment:
This isn't related to this PR, but instead of this (and your new config) being copied by value here, the normal pattern is to hold a shared_ptr to a Config object that has already converted the protobuf config to the internal represenation, and done validation such as no duplicates.